### ISSUE TITLE
Relative asset plugin broken

### ISSUE THREAD

**Issue by liaoyu1992** [nuxt/framework#3469]
### Environment

window10 
nodejs: v16.14.0

### Reproduction

GET http://localhost:3000/main-62bdb63b.mjs net::ERR_ABORTED 404 (Page not found: /main-62bdb63b.mjs)
GET http://localhost:3000/main.4f570753.css net::ERR_ABORTED 404 (Page not found: /main.4f570753.css)
entry-dfffb6f1.mjs:8 Event {isTrusted: true, type: 'error', target: link, currentTarget: link, eventPhase: 2, …}
Uncaught (in promise) Event {isTrusted: true, type: 'error', target: link, currentTarget: null, eventPhase: 0, …}

### Describe the bug

When I'm routing to a certain page,for example:
router.push({ name: 'main'}),
then report the above error

**Comment by phanmn** [comment]
Seeing this issue also

**Comment by danielroe** [comment]
Would you provide a reproduction? 🙏

**Comment by dualmacops** [comment]
I have the same problem, but it looks like it only happens on the / route, not in the others

**Comment by Teranode** [comment]
seems to happen on this commit precisely: https://github.com/nuxt/framework/commit/a0f8a44e10958035a8b8ba76becf5145d0c9dc98

**Comment by zynth17** [comment]
got the same issues in preview mode.

**Comment by xiao-e-yun** [comment]
hashes match is true
but miss buildAssetsDir or `_nuxt/` ( cdnURL work

**Comment by peteromano** [comment]
Same.

Doesn't look like hashes in my case; column (a layout), is loaded correctly initially then attempted to load again under the root (not _nuxt).. I also don't know why the entire app is completely broken from this either....

![image](https://user-images.githubusercontent.com/392387/156643607-d79b9395-5314-4ff3-b3b2-aa9455a0052e.png)

**Comment by peteromano** [comment]
is there a workaround for now or proposed fix to wait on?

**Comment by xiao-e-yun** [comment]
This is my temporary solution

`node ./deploy.mjs`

~/deploy.mjs
```Js
import fs from 'node:fs/promises';
import path from 'node:path';

const p = "/_nuxt/"

const __dirname = new URL(".",import.meta.url).pathname
const from = path.resolve(__dirname,".output/public"+p)

try {
  let files = await fs.readdir(from)
  files = files.filter(v=>v.startsWith("entry-") && v.endsWith(".mjs"))

  if (files.length) {
    const file = path.resolve(from,files[0])
    let data = await fs.readFile(file,{encoding:"utf-8"})

    const replace = []
    const regex = /import\("(.*?)\.mjs"\)/gm
    let match = undefined

    data = data.replaceAll("./","")

    for (;;) {
      match = regex.exec(data)
      if(match===null) break
      replace.push(match[1])
    }

    replace.forEach(v=>{
      const css = new RegExp(v.slice(0,v.indexOf("-"))+"\.(.{8})\.css")
      data = data.replaceAll(v+".mjs",path.resolve(p,v+".mjs"))
      data = data.replace(css,p+"$&")
    })
    await fs.writeFile(file,data)
  }
} catch(e) {
  console.error(e)
}
```

**Comment by lukeocodes** [comment]
> @peteromano for now you can use `3.0.0-27434608.0c99002` as the nuxt3 version until it's fixed

This release didn't resolve it for me

**Comment by bmhtech07** [comment]
I've experienced the same issue.

This worked for me today:

```
  "devDependencies": {
    "nuxt3": "3.0.0-27434608.0c99002",
  }
```

**Comment by danielroe** [comment]
@bmhtech07 This bug should not be present in the current version of Nuxt. If it is, would you open a new issue with a reproduction?

**Comment by bmhtech07** [comment]
False alarm. Staging branch hadn't been updated!  Now fine!

**Comment by ennioVisco** [comment]
> @bmhtech07 This bug should not be present in the current version of Nuxt. If it is, would you open a new issue with a reproduction?

It seems to be present again in `3.0.0-27480376.fdd38f9`. Rolling back to that commit solves it

**Comment by danielroe** [comment]
@ennioVisco Would you open a new issue with a reproduction?
